### PR TITLE
WIP: Add Cmake configuration for building code exercises

### PR DESCRIPTION
**Pull request by AerialMantis**
* Add Cmake configuration.
* Add Catch2, ComputeCpp SDK and Stb image as submodules.
* Add options for specifying the SYCL implementation.
* Move CppCon exercises into Code_Exercises root.
* Remove unnecessary CppCon files.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+endif()
+if (${SYCL_IMPLEMENTATIONS_USED} GREATER 1)
+  message(FATAL_ERROR "Multiple SYCL implementations specified, please only "
+  "set one of SYCL_ACADEMY_USE_COMPUTECPP, SYCL_ACADEMY_USE_DPCPP or "
+  "SYCL_ACADEMY_USE_HIPSYCL to ON.")
+endif()
+
+# Common setup
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/Cmake")
```
Comment (DuncanMcBain) [inline CMakeLists.txt:43]: you should append to this path, rather than set it (other modules will break)

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+include_directories(${CMAKE_SOURCE_DIR}/external/catch2)
```
Comment (DuncanMcBain) [inline CMakeLists.txt:45]: Can we use the [CMake support](https://github.com/catchorg/Catch2/blob/master/docs/cmake-integration.md#user-content-cmake-target) provided by Catch2?

File: Code_Exercises/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  SYCL Academy is licensed under a Creative Commons Attribution-ShareAlike 4.0
+  International License.
+
+  You should have received a copy of the license along with this work.  If not,
+  see <http://creativecommons.org/licenses/by-sa/4.0/>.
+]]
+
+# TODO(Gordon): Generalize this to support other implementations
+function(add_sycl_executable prefix source)
```
Comment (DuncanMcBain) [inline Code_Exercises/CMakeLists.txt:12]: The indentation has gone a bit funny in this function

File: CMakeLists.txt (hunk 1 above)
Comment (AerialMantis) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 2 above)
Comment (AerialMantis) [reply CMakeLists.txt]: Good idea, done.

File: Code_Exercises/CMakeLists.txt (hunk 3 above)
Comment (AerialMantis) [reply Code_Exercises/CMakeLists.txt:12]: Yup, fixed this.

**Comment by AerialMantis** [comment]
Thanks for the feedback @DuncanMcBain I've added Catch2 and ComputeCpp SDK as submodules and add the catch directory using the target.

File: Code_Exercises/CMakeLists.txt [hunk 4]
```diff
@@ -11,16 +11,17 @@
 # TODO(Gordon): Generalize this to support other implementations
 function(add_sycl_executable prefix source)
 	add_executable("${prefix}_${source}" "${source}.cpp")
-    target_include_directories("${prefix}_${source}" PRIVATE ${PROJECT_SOURCE_DIR}/include ${PROJECT_SOURCE_DIR}/external/catch2)
-	target_link_libraries("${prefix}_${source}" PRIVATE Threads::Threads)
-    set_target_properties("${prefix}_${source}" PROPERTIES CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON)
+  target_link_libraries("${prefix}_${source}" PRIVATE Threads::Threads)
+	target_link_libraries("${prefix}_${source}" PUBLIC Catch2::Catch2)
```
Comment (DuncanMcBain) [inline Code_Exercises/CMakeLists.txt:15]: I think there's still a tab on this line, rather than two spaces

Comment (AerialMantis) [reply Code_Exercises/CMakeLists.txt]: Fixed.

**Pull request merged**